[E2E/widgets-editor] Make sure the Welcome popup gets closed before the next step

WunderBart opened the pull request:
This PR ended up addressing 3 things:

1. **Widgets E2E suite flakiness**
  The flakiness was most probably caused by the fact that the Welcome guide's `Got it` button is initially _dead_ for a split second. To Playwright, that button is considered as clickable as it meets all the actionability conditions. But since the click was sometimes happening a bit _too early_, the "Welcome" banner sometimes wasn't disappearing and the test was getting stuck. It seems now to be resolved by waiting for the Widgets page to become stable instead of waiting for the `networkidle` event. [Here](https://github.com/Automattic/wp-calypso/pull/55760#issuecomment-907070093) are the stress test runs for the widgets suite.

2. **Sidebar flakiness on the `desktop` viewport**
  Sidebar flakiness on the desktop viewport was caused by the unstable appearance of the hovered item submenu. Playwright sometimes _notices_ that submenu and attempts to click it. This is a race condition because the submenu often disappears before the actual click action is performed, leaving the navigation in suspension. This seems now to be fixed by updating the submenu selector, which now explicitly targets the expanded submenu instead of targeting also the hovered one.

3. **Viewport name correction in the WebApp CI setup**
  Just [a small string correction](https://github.com/Automattic/wp-calypso/pull/55760/commits/4bfcbe1ed504a0cf756d8f2b3369d21b9599d971) - noticed that the viewport name is hardcoded in one spot.

WunderBart committed:
Make sure the Welcome popup gets closed before the next step
It seems like that popup is sometimes not actionable immediately despite Playwright claiming it is. The "Got it" button gets clicked but nothing happens. The added logic here makes sure the button is hidden before continuing to next step, and if not, if will repeat the click until it either succeeds or the global timeout is thrown.

fullofcaffeine commented on test/e2e/specs/specs-playwright/wp-widgets__spec.ts:
```diff
@@ -22,12 +22,27 @@ describe( DataHelper.createSuiteTitle( 'Widgets' ), function () {
+			try {
+				const button = await page.waitForSelector( buttonSelector );
+				await button.click();
+				// Retry if the button doesn't get hidden after 1 second. It can
```
Did you mean `popup` here instead of `button`?

WunderBart replied:
I meant button as it's what we're checking here but this is outdated now as I've changed the approach after some additional testing. See https://github.com/Automattic/wp-calypso/pull/55760/commits/0d802350415b4f7c142809a798849b7f878abd26.

WunderBart committed:
Wait for the widgets editor instead of click retry
Looks like when the menu is loaded & stable the Welcome popup is also actionable.

WunderBart committed:
[tmp] Stress-test wp-widgets__spec.js

WunderBart committed:
Display correct viewport in the "Run e2e tests" step

WunderBart committed:
Make sure there is no click attempt on the hovered submenu
This was the source of flakiness on desktop viewport. 😌

WunderBart committed:
Remove .hovered class dependancy
It should be either `.is-toggle-open` or `not:(.hovered)` - the first one is more straight forward to apply.

WunderBart committed:
Revert "[tmp] Stress-test wp-widgets__spec.js"

This reverts commit 28f8a30f9a977b818f021af231f00089ad326a61.
